components/post_view/channel_intro_message/add_members_button/add_members_button.tsx

```diff
@@ -51,27 +55,27 @@ const AddMembersButton: React.FC<AddMembersButtonProps> = ({totalUsers, usersLim
         >
             {inviteUsers && !isPrivate ? (
                 <LessThanMaxFreeUsers
-                    createBoard={createBoard}
-                    setHeader={setHeader}
+                    showBoardsButton={showBoardsButton}
+                    showSetHeader={showSetHeader}
                 />
             ) : (
                 <MoreThanMaxFreeUsers
                     channel={channel}
-                    createBoard={createBoard}
-                    setHeader={setHeader}
+                    showBoardsButton={showBoardsButton}
+                    showSetHeader={showSetHeader}
                 />
             )}
         </TeamPermissionGate>
     );
 };
 
-const LessThanMaxFreeUsers = ({setHeader, createBoard}: {setHeader: React.ReactNode; createBoard: React.ReactNode}) => {
+const LessThanMaxFreeUsers = ({showSetHeader, showBoardsButton}: {showSetHeader: boolean; showBoardsButton: boolean}) => {
     const {formatMessage} = useIntl();
 
     return (
         <>
-            {createBoard}
-            {setHeader}
+            <BoardsButton show={showBoardsButton}/>
+            <SetHeaderButton show={showSetHeader}/>
```

Comment:
I would prefer if we could remove the `show` prop and instead make this a conditional rendering

```suggestion
            {showBoardsButton && <BoardsButton/>}
            {showSetHeader && <SetHeaderButton/>}
```